packages/runtime/src/next/client.ts

```diff
@@ -140,33 +140,71 @@ export type MakeswiftPageSnapshot = {
 
 const makeswiftComponentDocumentSchema = z.object({
   id: z.string(),
-  type: z.string(),
-  key: z.string(),
   name: z.string().nullable(),
   locale: z.string().nullable(),
   data: Schema.element,
+  siteId: z.string(),
 })
 
 export type MakeswiftComponentDocument = z.infer<typeof makeswiftComponentDocumentSchema>
 
-export type MakeswiftComponentSnapshot = {
-  document: MakeswiftComponentDocument
-  cacheData: CacheData
-}
+const makeswiftComponentDocumentFallbackSchema = z.object({
+  id: z.string(),
+  locale: z.string().nullable(),
+  data: z.null(),
+  apiKeyPrefix: z.string(),
```

Comment:
I'm a bit confused why the fallback has an API key prefix

Reply:
Could we send the ID instead of attempting to add this to some sort of model?

Reply:
We need to pass the `siteId` to `deterministicUUID`. But since we only have the siteApiKey and not the `siteId` on the host (if the snapshot 404), we use the `siteApiKey` as a seed for `deterministicUUID`

Reply:
I guess what I'm trying to say is: why are we sending the inputs to the ID to the builder instead of computing it in the host and sending it the deterministic ID? Couldn't the `id` field be the deterministic ID instead of the user-provided one so just like the non-fallback document, the `id` is what the builder uses to identify it? I'm guessing there's logic in the builder right now to compute the stable ID that isn't necessary (since the host could do it)?